lib/active_model/serializer/adapter/json/fragment_cache.rb

```diff
@@ -1,16 +1,6 @@
-require 'active_model/serializer/adapter/fragment_cache'
-module ActiveModel
-  class Serializer
-    class Adapter
-      class Json < Adapter
-        class FragmentCache
+class ActiveModel::Serializer::Adapter::Json::FragmentCache
```

Comment:
Adapter is a class.  For now, adapter can't be a module.  Anyway, the module convention exists to make things work.  This change forces all non-loaded superclasses to be autoloaded when referenced. That's the only way to do it now.

If you think using an inherited hook is magical... Well, it's a pretty fundamental part of Ruby meta programming.  Rails uses these hooks all over the place. If you forget the hook exists, do you prefer the way this pr behaves?  I.e.  You can name your adapter whatever you want.

P.s.  I do intend to fix the name spacing issue after this is merged